src/access.rs

```diff
@@ -0,0 +1,339 @@
+
+		// We double-check that no inputs try to spend non-witness outputs. As we use a SegWit
+		// wallet descriptor this technically shouldn't ever happen, but better safe than sorry.
+		for input in &psbt.inputs {
+			if input.witness_utxo.is_none() {
+				log_error!(self.logger, "Tried to spend a non-witness funding output. This must not ever happen. Panicking!");
+				panic!("Tried to spend a non-witness funding output. This must not ever happen.");
+			}
+		}
+
+		if !locked_wallet.sign(&mut psbt, SignOptions::default())? {
+			return Err(Error::FundingTxCreationFailed);
+		}
+
+		Ok(psbt.extract_tx())
+	}
+
+	pub(crate) fn get_new_address(&self) -> Result<bitcoin::Address, Error> {
+		let address_info = self.wallet.lock().unwrap().get_address(AddressIndex::New)?;
+		Ok(address_info.address)
+	}
+}
+
+impl<D> FeeEstimator for LdkLiteChainAccess<D>
+where
+	D: BatchDatabase,
+{
+	fn get_est_sat_per_1000_weight(&self, confirmation_target: ConfirmationTarget) -> u32 {
+		let num_blocks = num_blocks_from_conf_target(confirmation_target);
+		let fallback_fee = fallback_fee_from_conf_target(confirmation_target);
+		self.blockchain
+			.estimate_fee(num_blocks)
+			.map_or(fallback_fee, |fee_rate| (fee_rate.fee_wu(1000) as u32).max(MIN_FEERATE)) as u32
+	}
+}
+
+impl<D> BroadcasterInterface for LdkLiteChainAccess<D>
+where
+	D: BatchDatabase,
+{
+	fn broadcast_transaction(&self, tx: &Transaction) {
+		match self.blockchain.broadcast(tx) {
+			Ok(_) => {}
+			Err(err) => log_error!(self.logger, "Failed to broadcast transaction: {}", err),
+		}
+	}
+}
+
+impl<D> Filter for LdkLiteChainAccess<D>
+where
+	D: BatchDatabase,
+{
+	fn register_tx(&self, txid: &Txid, _script_pubkey: &Script) {
+		self.queued_transactions.lock().unwrap().push(*txid);
+	}
+
+	fn register_output(&self, output: WatchedOutput) -> Option<(usize, Transaction)> {
+		self.queued_outputs.lock().unwrap().push(output);
+		return None;
+	}
+}
+
+impl<D> GetHeight for LdkLiteChainAccess<D>
+where
+	D: BatchDatabase,
+{
+	fn get_height(&self) -> Result<u32, bdk::Error> {
+		self.blockchain.get_height()
+	}
+}
+
+impl<D> GetBlockHash for LdkLiteChainAccess<D>
+where
+	D: BatchDatabase,
+{
+	fn get_block_hash(&self, height: u64) -> Result<BlockHash, bdk::Error> {
+		self.blockchain.get_block_hash(height)
+	}
+}
+
+impl<D> GetTx for LdkLiteChainAccess<D>
+where
+	D: BatchDatabase,
+{
+	fn get_tx(&self, txid: &Txid) -> Result<Option<Transaction>, bdk::Error> {
+		self.blockchain.get_tx(txid)
+	}
+}
+
+fn num_blocks_from_conf_target(confirmation_target: ConfirmationTarget) -> usize {
+	match confirmation_target {
+		ConfirmationTarget::Background => 12,
+		ConfirmationTarget::Normal => 6,
+		ConfirmationTarget::HighPriority => 3,
+	}
+}
+
+fn fallback_fee_from_conf_target(confirmation_target: ConfirmationTarget) -> u32 {
+	match confirmation_target {
+		ConfirmationTarget::Background => MIN_FEERATE,
+		ConfirmationTarget::Normal => 2000,
```

Comment:
will these fee_rates be part of ldk-lite config eventually ?

Reply:
This is up for discussion, especially what we'd allow to set and what not, see (https://github.com/lightningdevkit/ldk-lite/issues/17).